src/login/logind-dbus.c

```diff
@@ -2368,6 +2386,10 @@ static int method_do_shutdown_or_sleep(
         if (action == HANDLE_SLEEP) {
                 HandleAction selected;
 
+                r = cache_wakeup_count();
```

Comment:
i think "cache" is misleading. This is not a cache really (i.e. the point of it is that things may go out of sync with the kernel's own counter). it's just a saved value. hence maybe "save_wakeup_count()?"